cmd/compute-domain-daemon/main.go

```diff
@@ -154,60 +173,93 @@ func run(ctx context.Context, cancel context.CancelFunc, flags *Flags) error {
 		nodeName:               flags.nodeName,
 		podIP:                  flags.podIP,
 	}
+	klog.Infof("config: %v", config)
 
-	// Create and start controller to watch for compute domain updates and keep
-	// the IPs mapped in the nodes config file in sync
+	daemonCommandLine := []string{imexBinaryPath, "-c", imexConfigPath}
+	processManager := NewProcessManager(daemonCommandLine)
 	controller, err := NewController(config)
 	if err != nil {
 		return fmt.Errorf("error creating controller: %w", err)
 	}
 
-	// Start controller in a goroutine
-	errChan := make(chan error, 1)
+	var wg sync.WaitGroup
+
+	// Start controller in goroutine
+	wg.Add(1)
 	go func() {
-		err := controller.Run(ctx)
-		if err != nil {
+		defer wg.Done()
+		if err := controller.Run(ctx); err != nil {
+			klog.Errorf("controller failed, initiate shutdown: %s", err)
 			cancel()
 		}
-		errChan <- err
 	}()
 
-	// Wait until all nodes have joined the compute domain
-	nodes, err := controller.BlockUntilAllNodesJoinComputeDomain(ctx)
-	if err != nil {
-		return fmt.Errorf("error waiting for all nodes to join ComputeDomain: %w", err)
-	}
+	// Start processNodeUpdates() in goroutine
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := processNodeUpdates(controller, ctx, flags.cliqueID, processManager); err != nil {
```

Comment:
can we think of a better name for what is being called here? it doesn't imply that the process will be started by this.